src/Components/ArtComments.jsx

```diff
@@ -1,31 +1,45 @@
-import { useState, useEffect } from 'react';
-import CommentCard from './CommentCard';
-import AddNewComment from './AddNewComment';
+import { useState, useEffect } from "react";
+import CommentCard from "./CommentCard";
+import AddNewComment from "./AddNewComment";
+import { GetCommentsByArticleId } from "../api";
 
-const CommentsByArticleId = ({article_id}) => {
-    const [comments, setComments] = useState(null);
-  
-    useEffect(()=>{
-      fetch(`https://newsapp-api-project.onrender.com/api/articles/${article_id}/comments`)
-      .then((response)=>{
-          return response.json()
-      })
-      .then((result)=>{
-          setComments(result.commentsById)
-      })
-      .catch((error) => {
+const CommentsByArticleId = ({ article_id }) => {
+  const [comments, setComments] = useState(null);
+  const [newComment, setNewComment] = useState(false);
+
+  useEffect(() => {
+    if (newComment) {
+      GetCommentsByArticleId(article_id)
+        .then((result) => {
+          setComments(result.commentsById);
+        })
+        .catch((error) => {
+          console.error("Error fetching comments:", error);
+          setComments(null);
+        });
+    } else {
+        GetCommentsByArticleId(article_id)
+        .then((result) => {
+          setComments(result.commentsById);
+        })
+        .catch((error) => {
           console.error("Error fetching comments:", error);
           setComments(null);
         });
-    }, [article_id, comments])
+    }
+  }, [newComment]);
 
-    if (comments === null) return <p>No comments</p>
-    return (
-       <>
-        <CommentCard comments={comments}/>
-        {/* <AddNewComment setComments={setComments} article_id={article_id}/> */}
-        </>
-    )
-}
+  if (comments === null) return <p>No comments</p>;
```

Comment:
this doesn't work on the article with no comments (article 37) and the whole page breaks instead